tests/multinode.rs

```diff
@@ -389,28 +390,59 @@ fn test_multi_node_dynamic_network() {
         send_tx_and_retry_get_balance(&leader_data, &alice, &bob_pubkey, Some(1000)).unwrap();
     assert_eq!(leader_balance, 1000);
 
-    let validators: Vec<(NodeInfo, FullNode)> = (0..N)
+    let validators_per_thread = Arc::new(RwLock::new(Vec::new()));
+    let threads: Vec<_> = (0..10)
         .into_iter()
-        .map(|n| {
-            let keypair = KeyPair::new();
-            let validator = TestNode::new_localhost_with_pubkey(keypair.pubkey());
-            let rd = validator.data.clone();
-            //send some tokens to the new validator
-            let bal =
-                send_tx_and_retry_get_balance(&leader_data, &alice, &keypair.pubkey(), Some(500));
-            assert_eq!(bal, Some(500));
-            let val = FullNode::new(
-                validator,
-                false,
-                LedgerFile::Path(ledger_path.clone()),
-                Some(keypair),
-                Some(leader_data.contact_info.ncp),
-            );
-            info!("started[{}/{}] {:x}", n, N, rd.debug_id());
-            (rd, val)
+        .map(move |i| {
+            let validators_per_thread = validators_per_thread.clone();
+            let t = Builder::new()
+                .name("child".to_string())
+                .spawn(move || {
+                    println!("Hello, world {}!", i);
+                    let validators: Vec<(NodeInfo, FullNode)> = (0..N / 10)
+                        .into_iter()
+                        .map(|n| {
+                            let keypair = KeyPair::new();
+                            let validator = TestNode::new_localhost_with_pubkey(keypair.pubkey());
+                            let rd = validator.data.clone();
+                            //send some tokens to the new validator
+                            let bal = send_tx_and_retry_get_balance(
```

Comment:
I would do this part first, before starting the validators, make sure their keys are valid and have a balance

Reply:
You mean, transfer balance to all validators, before any of them is started?

Reply:
yea, basically do all the prep work before any of them are started so start should be fast